controller/workitem.go

```diff
@@ -255,20 +250,32 @@ func (c *WorkitemController) Delete(ctx *app.DeleteWorkitemContext) error {
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
-	authorized, err := authz.Authorize(ctx, wi.SpaceID.String())
+
+	// This checks if user is space admin or workitem creator or not
+	creator := wi.Fields[workitem.SystemCreator]
+	if creator == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, errs.New("work item doesn't have creator")))
+	}
+	authorized, err := c.isWorkitemCreatorOrSpaceOwner(ctx, wi.SpaceID, creator.(string), currentUserIdentityID.String())
 	if err != nil {
-		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+		forbidden, err := errors.IsForbiddenError(err)
+		if forbidden {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to delete the workitem"))
+
+		}
+		return err
 	}
 	if !authorized {
```

Comment:
@DhritiShikhar This if block is unreachable. 
The `isWorkitemCreatorOrSpaceOwner` returns either `false, err` or `true, nil`. Which means when `authorized=false` we will have  `err != nil` and the if block on line 200 will be executed.

I'd suggest we change `isWorkitemCreatorOrSpaceOwner` method to return only `error`. If access is allowed, we return nil otherwise the appropriate error.